src/App.js

```diff
@@ -17,29 +19,33 @@ class App extends Component {
   render() {
     const logIn = () => {
       fetch(`http://localhost:5000/users/`)
-      .then(res => res.json())
-      .then(json => this.setState({user: json, isLogged: true}))
+        .then(res => res.json())
+        .then(json => this.setState({ user: json, isLogged: true }))
     }
     const logOut = () => {
-      this.setState({user: {}, isLogged: false})
+      this.setState({ user: {}, isLogged: false })
     }
 
     return (
-      <Fragment>
-        <Navbar
-          logIn={logIn}
-          logOut={logOut}
-          isLogged={this.state.isLogged}
-          user={this.state.user} />
+      <Router>
+      <Navbar
+      logIn={logIn}
+      logOut={logOut}
+      isLogged={this.state.isLogged}
+      user={this.state.user} />
         {this.state.isLogged &&
           <Alert>
-            <User
-              user={this.state.user} />
-            <InvoicesList />
-            <NewInvoiceForm />
+            <Switch>
+              <Route exact path='/user' render={(props) => <User user={this.state.user} />} />
+              <Route exact path='/invoices'>
+                <InvoicesList />
+              </Route>
+              <Route exact path='/invoices/new' component={NewInvoiceForm} />
+              <Route exact path='/invoices/:id' component={InvoiceDetails} />
+              <Route exact path='/contractor/new' component={NewContractorForm} />
+            </Switch>
           </Alert>}
-
-      </Fragment>
+      </Router>
```

Comment:
FYI: On larger projects, where you have multiple providers, a good practice is to keep routing in separate `<Routes>` component. For your purpose this is ok.